File: Blind75/ruby/array/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hash = {}
    len = nums.length - 1
    for i in(0..len) 
        diff = target - nums[i] 
        if hash.include?(diff) 
            return [i, hash[diff]] 
        end  
        hash[nums[i]] = i  
    end 
    return 
end

File: Blind75/java/array/TwoSum.java
package Blind75.java.array;

import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) { 
        int len = nums.length;
        HashMap<Integer, Integer> map = new HashMap<>(len); 
        for(int i = 0; i < len; i++) { 
            int diff = target - nums[i]; 
            if(map.containsKey(diff)) { 
                return new int[]{i, map.get(diff)};
            }
            map.put(nums[i], i);
        }  
       return new int[0];
    }
}
